src/domain/order/repositories/product.ts

```diff
@@ -9,4 +9,5 @@ export interface ProductRepository {
     name: string,
   ): Promise<Product | null>;
   findById(id: string): Promise<Product | null>;
+  findByCategory(category: string): Promise<Product[] | null>;
```

Comment:
```suggestion
  findByCategory(category: string): Promise<Product[]>;
```

Esse cara é meio que uma listagem então deve sempre retornar uma lista, mesmo que vazia